### DISCUSSION TITLE
PHOTOPRISM_DISABLE_TLS environment variable not used?

### DISCUSSION THREAD

**Discussion by barry-luijten** [photoprism/photoprism#3819] [General]
Hi,

I have just upgraded Photoprism to the latest version (231011) and I noticed these logs during startup of the container, while I explicitly set the environment variables PHOTOPRISM_DISABLE_TLS="true" and PHOTOPRISM_DEFAULT_TLS="false":
```
started 231011 as root (amd64-prod)
init: https
Creating a default HTTPS/TLS certificate.
Generating self-signed issuer (CA) certificate...
Generating self-signed tls certificate...
Certificate request self-signature ok
subject=C = DE, ST = Berlin, L = Berlin, O = PhotoPrism, OU = Self-Hosted, emailAddress = hello@photoprism.app, CN = PhotoPrism
Updating permissions of keys in '/etc/ssl/private'...
Running 'update-ca-certificates'...
Updating certificates in /etc/ssl/certs...
mv: cannot move '/etc/ssl/certs/ca-certificates.crt.new' to 'ca-certificates.crt': Device or resource busy
make: *** [Makefile:16: install-https] Error 1
```
The error message is caused by the fact that I am mounting my own ca-certificates from a Docker config, so it's read-only in the container. I just think it is strange that Photoprism seems to be generating a self-signed CA and certificate while I configured it not to to anything with TLS. Photoprism seems to be working fine, but it looks like the environment variables have no effect on this behavior.
FYI, I'm using Traefik Proxy for handling https certificates, so no need for them in the container.

Please let me know if I'm doing anything wrong here ;)

**Comment by lastzero** [comment]
These logs are created by the entrypoint script before PhotoPrism starts, e.g. because the `PHOTOPRISM_INIT` environment variable default has not been changed or it contains "https":
- https://docs.photoprism.app/getting-started/config-options/#docker-image

**Reply by barry-luijten** [reply]
Ah, I see! I'm not explicitly setting PHOTOPRISM_INIT, so the default must contain something like https then. I'll check later if that is the case and if so, I will empty it explicitly in my yml. Thanks for the answer!

**Reply by barry-luijten** [reply]
Just to confirm: setting ```PHOTOPRISM_INIT``` to an empty value (or by removing https, which is the default, built-in value of the docker image), removes the above startup log messages 👍🏻

**Reply by lastzero** [reply]
I've improved our script so that it skips the creation of a self-signed certificate if `PHOTOPRISM_DISABLE_TLS="false"`:
- https://github.com/photoprism/photoprism/issues/3823
